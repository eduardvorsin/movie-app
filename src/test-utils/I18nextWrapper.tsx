import React, { ReactNode } from 'react';
import i18next from '@/i18n/client';
import { I18nextProvider } from 'react-i18next';

type Props = {
	children: ReactNode,
}

const I18nextWrapper = ({
	children,
}: Props) => {
	return (
		<I18nextProvider i18n={i18next}>
			{children}
		</I18nextProvider>
	);
};

export default I18nextWrapper;