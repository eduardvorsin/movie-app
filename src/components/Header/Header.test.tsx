import { render, screen } from "@testing-library/react";
import Header from "./Header";
import i18next from "@/i18n/client";
import I18nextWrapper from "@/test-utils/I18nextWrapper";

const mockUseSelectedLayoutSegments = jest.fn();
jest.mock('next/navigation', () => ({
	useSelectedLayoutSegments: () => mockUseSelectedLayoutSegments(),
}));

describe('Header tests', () => {
	it('is rendered correctly', () => {
		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		const dictionary = {
			searchButton: 'To find',
			logo: {
				linkText: 'To home page',
				altText: 'MovieWander logo'
			},
			navigation: {
				movies: 'movies',
				persons: 'persons',
				tv: 'series and shows',
				new: 'new releases',
				collections: 'collections'
			},
			userSettingsButton: {
				button: 'Open settings',
				themeTitle: 'The theme of the site',
				languageTitle: 'Language Settings',
				themeToggle: { label: 'Light and dark theme switch' },
				languageSelect: { label: 'Language selection' },
			},
			navigationMenuButton: {
				active: 'Close the navigation menu',
				inactive: 'Open the navigation menu',
			}
		};

		render(
			<Header
				dictionary={dictionary}
				testId='test-header'
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-header')).toBeInTheDocument();
	});

	it('is a basic snapshot', () => {
		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		const dictionary = {
			searchButton: 'To find',
			logo: {
				linkText: 'To home page',
				altText: 'MovieWander logo'
			},
			navigation: {
				movies: 'movies',
				persons: 'persons',
				tv: 'series and shows',
				new: 'new releases',
				collections: 'collections'
			},
			userSettingsButton: {
				button: 'Open settings',
				themeTitle: 'The theme of the site',
				languageTitle: 'Language Settings',
				themeToggle: { label: 'Light and dark theme switch' },
				languageSelect: { label: 'Language selection' },
			},
			navigationMenuButton: {
				active: 'Close the navigation menu',
				inactive: 'Open the navigation menu',
			}
		};
		render(
			<Header
				dictionary={dictionary}
				testId='test-header'
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-header')).toMatchSnapshot();
	});
});

describe('Header integration tests', () => {
	it('localization into English works correctly', async () => {
		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		await i18next.changeLanguage('en');
		const translation = new RegExp(i18next.t('header.searchButton'));

		const dictionary = {
			searchButton: i18next.t('header.searchButton'),
			userSettingsButton: {
				button: i18next.t('userSettingsButton.button'),
				themeTitle: i18next.t('userSettingsButton.themeTitle'),
				languageTitle: i18next.t('userSettingsButton.languageTitle'),
				languageSelect: { label: i18next.t('languageSelect.label') },
				themeToggle: { label: i18next.t('themeToggle.label') },
			},
			navigationMenuButton: {
				active: i18next.t('navigationMenuButton.active'),
				inactive: i18next.t('navigationMenuButton.inactive'),
			},
			logo: {
				altText: i18next.t('logo.altText'),
				linkText: i18next.t('logo.linkText'),
			},
			navigation: {
				movies: i18next.t('navigation.movies'),
				persons: i18next.t('navigation.persons'),
				tv: i18next.t('navigation.tv'),
				new: i18next.t('navigation.new'),
				collections: i18next.t('navigation.collections'),
			},
		};
		render(
			<Header
				dictionary={dictionary}
				testId='test-header'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLHeadingElement>(translation)).toBeInTheDocument();
	});

	it('localization into Russian works correctly', async () => {
		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		await i18next.changeLanguage('ru');
		const translation = new RegExp(i18next.t('header.searchButton'));

		const dictionary = {
			searchButton: i18next.t('header.searchButton'),
			userSettingsButton: {
				button: i18next.t('userSettingsButton.button'),
				themeTitle: i18next.t('userSettingsButton.themeTitle'),
				languageTitle: i18next.t('userSettingsButton.languageTitle'),
				languageSelect: { label: i18next.t('languageSelect.label') },
				themeToggle: { label: i18next.t('themeToggle.label') },
			},
			navigationMenuButton: {
				active: i18next.t('navigationMenuButton.active'),
				inactive: i18next.t('navigationMenuButton.inactive'),
			},
			logo: {
				altText: i18next.t('logo.altText'),
				linkText: i18next.t('logo.linkText'),
			},
			navigation: {
				movies: i18next.t('navigation.movies'),
				persons: i18next.t('navigation.persons'),
				tv: i18next.t('navigation.tv'),
				new: i18next.t('navigation.new'),
				collections: i18next.t('navigation.collections'),
			},
		};
		render(
			<Header
				dictionary={dictionary}
				testId='test-header'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLHeadingElement>(translation)).toBeInTheDocument();
	});

	it('is a snapshot with English localization', async () => {
		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		await i18next.changeLanguage('en');

		const dictionary = {
			searchButton: i18next.t('header.searchButton'),
			userSettingsButton: {
				button: i18next.t('userSettingsButton.button'),
				themeTitle: i18next.t('userSettingsButton.themeTitle'),
				languageTitle: i18next.t('userSettingsButton.languageTitle'),
				languageSelect: { label: i18next.t('languageSelect.label') },
				themeToggle: { label: i18next.t('themeToggle.label') },
			},
			navigationMenuButton: {
				active: i18next.t('navigationMenuButton.active'),
				inactive: i18next.t('navigationMenuButton.inactive'),
			},
			logo: {
				altText: i18next.t('logo.altText'),
				linkText: i18next.t('logo.linkText'),
			},
			navigation: {
				movies: i18next.t('navigation.movies'),
				persons: i18next.t('navigation.persons'),
				tv: i18next.t('navigation.tv'),
				new: i18next.t('navigation.new'),
				collections: i18next.t('navigation.collections'),
			},
		};
		render(
			<Header
				dictionary={dictionary}
				testId='test-header'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByTestId<HTMLDivElement>('test-header')).toMatchSnapshot();
	});

	it('is a snapshot with Russian localization', async () => {
		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		await i18next.changeLanguage('ru');

		const dictionary = {
			searchButton: i18next.t('header.searchButton'),
			userSettingsButton: {
				button: i18next.t('userSettingsButton.button'),
				themeTitle: i18next.t('userSettingsButton.themeTitle'),
				languageTitle: i18next.t('userSettingsButton.languageTitle'),
				languageSelect: { label: i18next.t('languageSelect.label') },
				themeToggle: { label: i18next.t('themeToggle.label') },
			},
			navigationMenuButton: {
				active: i18next.t('navigationMenuButton.active'),
				inactive: i18next.t('navigationMenuButton.inactive'),
			},
			logo: {
				altText: i18next.t('logo.altText'),
				linkText: i18next.t('logo.linkText'),
			},
			navigation: {
				movies: i18next.t('navigation.movies'),
				persons: i18next.t('navigation.persons'),
				tv: i18next.t('navigation.tv'),
				new: i18next.t('navigation.new'),
				collections: i18next.t('navigation.collections'),
			},
		};
		render(
			<Header
				dictionary={dictionary}
				testId='test-header'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByTestId<HTMLDivElement>('test-header')).toMatchSnapshot();
	});
});
