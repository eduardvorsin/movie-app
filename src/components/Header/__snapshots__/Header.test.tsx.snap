// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header integration tests is a snapshot with English localization 1`] = `
<header
  class="pr-[var(--scrollbar-width)] fixed z-500 top-0 left-0 w-full h-[60px] bg-neutral-300 dark:bg-dark-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-header"
>
  <div
    class="max-w-[77rem] mx-auto px-4 flex items-center h-full"
  >
    <a
      class="inline-flex z-100 mr-10"
      href="/"
    >
      <img
        alt="MovieWander logo"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="http://localhost/assets/images/logo.svg"
        style="color: transparent;"
        width="62"
      />
      <span
        class="sr-only"
      >
        To home page
      </span>
    </a>
    <div
      class="fixed inset-0 bg-neutral-100/90 dark:bg-dark-neutral-100/90 z-0 md:static md:inset-auto md:bg-transparent dark:md:bg-transparent mt-[3.75rem] md:mt-0 md:translate-x-0 transition-colors transition-transform duration-300 -translate-x-full md:mr-10"
    >
      <nav
        class="z-100 w-full p-6 pt-3 md:p-0 bg-neutral-200 dark:bg-dark-neutral-250 md:bg-transparent dark:md:bg-transparent flex flex-col md:flex-row transition-colors duration-150"
      >
        <ul
          class="flex md:flex-wrap md:gap-2 flex-col md:flex-row"
        >
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-100"
              href="/movies"
            >
              movies
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/persons"
            >
              persons
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/tv"
            >
              series and shows
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/new"
            >
              new releases
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/collections"
            >
              collections
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="flex flex-grow justify-end"
    >
      <search
        class=" flex items-center justify-end flex-grow  "
        role="search"
      >
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   mr-1"
        >
          <span
            class="inline-flex items-center"
          >
            <svg
              class="p-2 w-[2.25rem] h-[2.25rem]"
              viewBox="0 0 32 32"
            >
              <use
                href="/assets/icons/search.svg#search"
              />
            </svg>
            Show the search field
          </span>
        </button>
      </search>
      <div
        class="relative w-[2.25rem] h-[2.25rem]"
        id=":r4:"
      >
        <button
          aria-controls=":r4:"
          aria-expanded="false"
          aria-haspopup="menu"
          aria-owns=":r4:"
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   "
        >
          <span
            class="inline-flex items-center"
          >
            <svg
              class="p-2 w-[2.25rem] h-[2.25rem] fill-current"
              viewBox="0 0 32 32"
            >
              <use
                href="/assets/icons/settings.svg#settings"
              />
            </svg>
            Open settings
          </span>
        </button>
      </div>
      <button
        class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   relative z-100 md:hidden"
      >
        <span
          class="inline-flex items-center"
        >
          <svg
            class="p-2 w-[2.25rem] h-[2.25rem]"
            fill="none"
            viewBox="0 0 32 32"
          >
            <path
              class="origin-center transition-transform duration-150  translate-x-0 translate-y-0 rotate-0 scale-100"
              d="M2 7H30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="3"
            />
            <path
              class="origin-center transition-opacity duration-150  opacity-100"
              d="M2 15.6667H30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="3"
            />
            <path
              class="origin-center transition-transform duration-150  translate-x-0 translate-y-0 rotate-0 scale-100"
              d="M2 24.3333H30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="3"
            />
          </svg>
          Open the navigation menu
        </span>
      </button>
    </div>
  </div>
</header>
`;

exports[`Header integration tests is a snapshot with Russian localization 1`] = `
<header
  class="pr-[var(--scrollbar-width)] fixed z-500 top-0 left-0 w-full h-[60px] bg-neutral-300 dark:bg-dark-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-header"
>
  <div
    class="max-w-[77rem] mx-auto px-4 flex items-center h-full"
  >
    <a
      class="inline-flex z-100 mr-10"
      href="/"
    >
      <img
        alt="MovieWander логотип"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="http://localhost/assets/images/logo.svg"
        style="color: transparent;"
        width="62"
      />
      <span
        class="sr-only"
      >
        На главную страницу
      </span>
    </a>
    <div
      class="fixed inset-0 bg-neutral-100/90 dark:bg-dark-neutral-100/90 z-0 md:static md:inset-auto md:bg-transparent dark:md:bg-transparent mt-[3.75rem] md:mt-0 md:translate-x-0 transition-colors transition-transform duration-300 -translate-x-full md:mr-10"
    >
      <nav
        class="z-100 w-full p-6 pt-3 md:p-0 bg-neutral-200 dark:bg-dark-neutral-250 md:bg-transparent dark:md:bg-transparent flex flex-col md:flex-row transition-colors duration-150"
      >
        <ul
          class="flex md:flex-wrap md:gap-2 flex-col md:flex-row"
        >
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-100"
              href="/movies"
            >
              фильмы
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/persons"
            >
              люди
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/tv"
            >
              сериалы и шоу
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/new"
            >
              новинки
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/collections"
            >
              коллекции
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="flex flex-grow justify-end"
    >
      <search
        class=" flex items-center justify-end flex-grow  "
        role="search"
      >
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   mr-1"
        >
          <span
            class="inline-flex items-center"
          >
            <svg
              class="p-2 w-[2.25rem] h-[2.25rem]"
              viewBox="0 0 32 32"
            >
              <use
                href="/assets/icons/search.svg#search"
              />
            </svg>
            Показать поле поиска
          </span>
        </button>
      </search>
      <div
        class="relative w-[2.25rem] h-[2.25rem]"
        id=":r5:"
      >
        <button
          aria-controls=":r5:"
          aria-expanded="false"
          aria-haspopup="menu"
          aria-owns=":r5:"
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   "
        >
          <span
            class="inline-flex items-center"
          >
            <svg
              class="p-2 w-[2.25rem] h-[2.25rem] fill-current"
              viewBox="0 0 32 32"
            >
              <use
                href="/assets/icons/settings.svg#settings"
              />
            </svg>
            Открыть настройки
          </span>
        </button>
      </div>
      <button
        class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   relative z-100 md:hidden"
      >
        <span
          class="inline-flex items-center"
        >
          <svg
            class="p-2 w-[2.25rem] h-[2.25rem]"
            fill="none"
            viewBox="0 0 32 32"
          >
            <path
              class="origin-center transition-transform duration-150  translate-x-0 translate-y-0 rotate-0 scale-100"
              d="M2 7H30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="3"
            />
            <path
              class="origin-center transition-opacity duration-150  opacity-100"
              d="M2 15.6667H30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="3"
            />
            <path
              class="origin-center transition-transform duration-150  translate-x-0 translate-y-0 rotate-0 scale-100"
              d="M2 24.3333H30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="3"
            />
          </svg>
          Открыть меню навигации
        </span>
      </button>
    </div>
  </div>
</header>
`;

exports[`Header tests is a basic snapshot 1`] = `
<header
  class="pr-[var(--scrollbar-width)] fixed z-500 top-0 left-0 w-full h-[60px] bg-neutral-300 dark:bg-dark-neutral-300 transition-colors duration-150 undefined"
  data-testid="test-header"
>
  <div
    class="max-w-[77rem] mx-auto px-4 flex items-center h-full"
  >
    <a
      class="inline-flex z-100 mr-10"
      href="/"
    >
      <img
        alt="MovieWander logo"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="http://localhost/assets/images/logo.svg"
        style="color: transparent;"
        width="62"
      />
      <span
        class="sr-only"
      >
        To home page
      </span>
    </a>
    <div
      class="fixed inset-0 bg-neutral-100/90 dark:bg-dark-neutral-100/90 z-0 md:static md:inset-auto md:bg-transparent dark:md:bg-transparent mt-[3.75rem] md:mt-0 md:translate-x-0 transition-colors transition-transform duration-300 -translate-x-full md:mr-10"
    >
      <nav
        class="z-100 w-full p-6 pt-3 md:p-0 bg-neutral-200 dark:bg-dark-neutral-250 md:bg-transparent dark:md:bg-transparent flex flex-col md:flex-row transition-colors duration-150"
      >
        <ul
          class="flex md:flex-wrap md:gap-2 flex-col md:flex-row"
        >
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-100"
              href="/movies"
            >
              movies
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/persons"
            >
              persons
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/tv"
            >
              series and shows
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/new"
            >
              new releases
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
              href="/collections"
            >
              collections
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="flex flex-grow justify-end"
    >
      <search
        class=" flex items-center justify-end flex-grow  "
        role="search"
      >
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   mr-1"
        >
          <span
            class="inline-flex items-center"
          >
            <svg
              class="p-2 w-[2.25rem] h-[2.25rem]"
              viewBox="0 0 32 32"
            >
              <use
                href="/assets/icons/search.svg#search"
              />
            </svg>
            Show the search field
          </span>
        </button>
      </search>
      <div
        class="relative w-[2.25rem] h-[2.25rem]"
        id=":r1:"
      >
        <button
          aria-controls=":r1:"
          aria-expanded="false"
          aria-haspopup="menu"
          aria-owns=":r1:"
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   "
        >
          <span
            class="inline-flex items-center"
          >
            <svg
              class="p-2 w-[2.25rem] h-[2.25rem] fill-current"
              viewBox="0 0 32 32"
            >
              <use
                href="/assets/icons/settings.svg#settings"
              />
            </svg>
            Open settings
          </span>
        </button>
      </div>
      <button
        class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-neutral-600 enabled:hover:text-neutral-700 enabled:active:text-neutral-800 dark:text-neutral-300 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100   relative z-100 md:hidden"
      >
        <span
          class="inline-flex items-center"
        >
          <svg
            class="p-2 w-[2.25rem] h-[2.25rem]"
            fill="none"
            viewBox="0 0 32 32"
          >
            <path
              class="origin-center transition-transform duration-150  translate-x-0 translate-y-0 rotate-0 scale-100"
              d="M2 7H30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="3"
            />
            <path
              class="origin-center transition-opacity duration-150  opacity-100"
              d="M2 15.6667H30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="3"
            />
            <path
              class="origin-center transition-transform duration-150  translate-x-0 translate-y-0 rotate-0 scale-100"
              d="M2 24.3333H30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="3"
            />
          </svg>
          Open the navigation menu
        </span>
      </button>
    </div>
  </div>
</header>
`;
